src/cli.js

```diff
@@ -611,15 +611,24 @@ var exports = {
     }
 
     var filename;
+    var lintStdinFile;
 
     // There is an if(filename) check in the lint() function called below.
     // passing a filename of undefined is the same as calling the function
     // without a filename.  If there is no opts.filename, filename remains
     // undefined and lint() is effectively called with 4 parameters.
     if (opts.filename) {
-      filename = path.resolve(opts.filename);
+      filename = opts.filename;
+      var ignores = !opts.ignores ? loadIgnores({ cwd: opts.cwd }) :
```

Comment:
`opts.ignores` is always a truthy value, so this ternary doesn't seem necessary.

Reply:
This new code is duplicated from the `gathers` function at https://github.com/jshint/jshint/blob/963d38d4c4f1aa31c373b64dbf08820f211b998b/src/cli.js#L574, so I think any fix should probably be made in both places.

Just to check, am I correct in thinking that the code in line 769 means that `opts.ignores` will always be `loadIgnores({ exclude: options.exclude, excludePath: options["exclude-path"] })`, and so therefore the ternary is unneccessary, or is there a way to call into this function that doesn't set `opts.ignores`?

Reply:
That's correct. It'll be good to clean up `gather`, but please do so in a dedicated commit to differentiate the refactoring from the bug fix.

Reply:
Since it is duplicated, I am going to not change either place for now